# HG changeset patch
# User Rik <rik@octave.org>
# Date 1541901157 28800
#      Sat Nov 10 17:52:37 2018 -0800
# Node ID 6ceffae58e0c33b44b29c308740cb4671135391f
# Parent  966d7fd88bb96a6e73ea8455f247e63a82742df3
# Parent  6250060e9e04cbc1a066a4d5c6f1d9d097b1d369
maint: merge stable to default.

diff --git a/scripts/plot/draw/plotyy.m b/scripts/plot/draw/plotyy.m
--- a/scripts/plot/draw/plotyy.m
+++ b/scripts/plot/draw/plotyy.m
@@ -118,17 +118,17 @@ function [ax, h1, h2] = __plotyy__ (ax, 
     set (ax(1), "ycolor", getcolor (h1));
   endif
 
   set (gcf (), "nextplot", "add");
 
   axes (ax(2));
 
   colors = get (ax(1), "colororder");
-  set (ax(2), "colororder", [colors(2:end,:); colors(1,:)]);
+  set (ax(2), "colororder", circshift (colors, -numel (h1), 1));
 
   if (strcmp (get (ax(1), "__autopos_tag__"), "subplot"))
     set (ax(2), "__autopos_tag__", "subplot");
   else
     set (ax, "activepositionproperty", "position");
   endif
 
   ## Don't replace axis which has colororder property already modified
